sway/desktop/render.c

```diff
@@ -235,13 +451,82 @@ static void render_drag_icons(struct sway_output *output,
 		render_surface_iterator, &data);
 }
 
+void render_whole_output(struct fx_renderer *renderer, pixman_region32_t *original_damage,
+		struct fx_texture *texture) {
+	struct wlr_output *output = renderer->sway_output->wlr_output;
+	struct wlr_box monitor_box = get_monitor_box(output);
+
+	enum wl_output_transform transform =
+		wlr_output_transform_invert(output->transform);
+	float matrix[9];
+	wlr_matrix_project_box(matrix, &monitor_box, transform, 0.0,
+		output->transform_matrix);
+
+	pixman_region32_t damage;
+	pixman_region32_init(&damage);
+	pixman_region32_union_rect(&damage, &damage, monitor_box.x, monitor_box.y,
+		monitor_box.width, monitor_box.height);
+	pixman_region32_intersect(&damage, &damage, original_damage);
+	bool damaged = pixman_region32_not_empty(&damage);
+	if (!damaged) {
+		goto damage_finish;
+	}
+
+	int nrects;
+	pixman_box32_t *rects = pixman_region32_rectangles(&damage, &nrects);
+	for (int i = 0; i < nrects; ++i) {
+		scissor_output(output, &rects[i]);
+		fx_render_texture_with_matrix(renderer, texture, &monitor_box, matrix, get_undecorated_decoration_data());
+	}
+
+damage_finish:
+	pixman_region32_fini(&damage);
+}
+
+void render_monitor_blur(struct sway_output *output, pixman_region32_t *damage) {
+	struct wlr_output *wlr_output = output->wlr_output;
+	struct fx_renderer *renderer = output->renderer;
+
+	struct wlr_box monitor_box = get_monitor_box(wlr_output);
+	scale_box(&monitor_box, wlr_output->scale);
```

Comment:
do we need to scale if we already have a box of the whole monitor? We don't scale this box elsewhere

Reply:
Nope. Broke scaling xd.

f1b6f1f7 fixes